thunder/tests/test_fa3_executor.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+import torch
+
+import thunder
+import thunder.core
+from thunder.executors.fa3ex import fa3_ex
+from thunder.tests.framework import requiresCUDA
+
+# flash attn 3
+try:
+    from flash_attn_interface import _flash_attn_forward, _flash_attn_backward, flash_attn_func
+
+    HAS_FA3 = True
+except:
+    HAS_FA3 = False
+
+
+@pytest.mark.parametrize("device,", ["cuda"])
+@pytest.mark.parametrize("dtype", [torch.float16], ids=("float16"))
+@requiresCUDA
+def test_fa3(device: str, dtype: torch.dtype):
+
```

Comment:
Would you add a device check for `fa3`?
For example:
```python
    # Flash Attention 3 is only supported on Hopper devices.
    # Skip this test on Ampere and prior devices.
    torch_device = torch.device("cuda")
    if not device_version_support(torch_device, CudaVersion(9, 0)):
        pytest.skip(f"flash attention 3 is not supported on {torch.cuda.get_device_name()}")
```

Reply:
For test_fa3 specifically, I already have an if not HAS_FA3 pytest.skip(), which should already cover the case of non-Hopper devices, since HAS_FA3 will be true if and only if the device is Hopper and the user / container has built fa3.